pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -53,45 +53,43 @@ class ParametrizedHamiltonian:
 
     """
 
-    num_wires = qml.operation.AnyWires
-    grad_method = "A"  # supports analytic gradients
-
-    def __init__(
-        self,
-        coeffs,
-        observables,
-    ):
+    def __init__(self, coeffs, observables):
 
         if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        for obs in observables:
-            if not isinstance(obs, Observable):
-                raise ValueError(
-                    "Could not create circuits. Some or all observables are not valid."
-                )
+        # ToDo: assuming this is how we want to do things, make this clear in the documentation
+        # since organizing into H_fixed and H_parametrized potentially moves around the order of the terms compared
+        # to how the terms were entered, sorting seems the least ambiguous organization for the wires
+        self.wires = Wires.all_wires([op.wires for op in observables], sort=True)
 
         self._ops = list(observables)
         self._coeffs = coeffs
 
-        self.H_fixed_coeffs = []
-        self.H_fixed_ops = []
-        self.H_parametrized_fns = []
-        self.H_parametrized_ops = []
+        self.H_coeffs_fixed = []
+        self.H_coeffs_parametrized = []
+        self.H_ops_fixed = []
+        self.H_ops_parametrized = []
 
         for coeff, obs in zip(coeffs, observables):
             if callable(coeff):
-                self.H_parametrized_fns.append(coeff)
-                self.H_parametrized_ops.append(obs)
+                self.H_coeffs_parametrized.append(coeff)
+                self.H_ops_parametrized.append(obs)
             else:
-                self.H_fixed_coeffs.append(coeff)
-                self.H_fixed_ops.append(obs)
+                self.H_coeffs_fixed.append(coeff)
+                self.H_ops_fixed.append(obs)
 
-    def __call__(self, params, t):
-        return self.H_fixed + self.H_parametrized(params, t)
+    def __call__(self, params, t, wires=None):
```

Comment:
Why do we want a `wires` argument?

Reply:
This gives the advantage to construct one `ParametrizedHamiltonian` to get multiple gates :)

Reply:
Ah yes. I think the idea was to provide users with the `wires` argument and perform the `map_wires` for them, basically as a convenience. Also see the Notion under `Final UI > Parametrized Hamiltonian > Wires argument`

Reply:
We had a similar discussion with @albi3ro when creating the `qml.map_wires` function.

We arrived to the conclusion that it might be confusing to the user when providing a list of wires. Imagine you have a `ParametrizedHamiltonian` with:

```pycon
>>> ops = [qml.PauliY(6), qml.PauliX(2), qml.PauliZ(4), qml.PauliY(3)]
>>> coeffs = [1 for _ in ops]
>>> H = ParametrizedHamiltonian(coeffs, ops)
>>> H([], 0)
(1*(PauliY(wires=[6]))) + (1*(PauliX(wires=[2]))) + (1*(PauliZ(wires=[4]))) + (1*(PauliY(wires=[3])))
```

To change the wires of this operator, it will be much easier to provide a dictionary mapping old wires to new wires than remembering the wire order and providing a list with the new wires. Let's say we want to map:
```pycon
>>> wire_map = {2: 10, 3: 11, 4: 12, 5: 13, 6: 14}
>>> new_wires = [14, 10, 12, 11]  # to build this list I had to read ALL the operators inside H (which can get tedious for big hamiltonians)
>>> H([], 0, wires=new_wires)
(1*(PauliY(wires=[14]))) + (1*(PauliX(wires=[10]))) + (1*(PauliZ(wires=[12]))) + (1*(PauliY(wires=[11])))
```

Reply:
I see, that makes sense! :+1: maybe worth pinging @trbromley here :)

Reply:
Happy to drop the `wires` argument from here for now (we can always add back in)